test/evp_test.c

```diff
@@ -3283,6 +3287,27 @@ static char *take_value(PAIR *pp)
     return p;
 }
 
+static int securitycheck_enabled(void)
+{
+    if (OSSL_PROVIDER_available(libctx, "fips")) {
+        OSSL_PARAM params[2];
+        OSSL_PROVIDER *prov = NULL;
+        int check = 1;
+
+        prov = OSSL_PROVIDER_load(libctx, "fips");
+        if (prov != NULL) {
+            params[0] =
+                OSSL_PARAM_construct_int(OSSL_PROV_PARAM_SECURITY_CHECKS,
+                                         &check);
+            params[1] = OSSL_PARAM_construct_end();
+            OSSL_PROVIDER_get_params(prov, params);
+            OSSL_PROVIDER_unload(prov);
+        }
+        return check;
```

Comment:
Can we not cache this value, rather than load and unload the provider every time....that seems quite heavy.

Reply:
static int coming up :)